pip_audit/dependency_source/resolvelib/pypi_provider.py

```diff
@@ -111,6 +122,37 @@ def get_metadata_for_wheel(url):
     return EmailMessage()  # pragma: no cover
 
 
+def get_metadata_for_sdist(url):
+    data = requests.get(url).content
+    metadata = EmailMessage()
+
+    with TemporaryDirectory() as pkg_dir:
+        # Extract archive onto the disk
+        with TarFile.open(fileobj=BytesIO(data), mode="r:gz") as t:
+            # The directory is the first member in a tarball
+            names = t.getnames()
+            pkg_name = names[0]
+            t.extractall(pkg_dir)
+
+        # Put together a full path of where the source distribution is
+        pkg_path = os.path.join(pkg_dir, pkg_name)
+
+        with TemporaryDirectory() as ve_dir:
+            ve = VirtualEnvironment(ve_dir)
+            ve.install(f"-e {pkg_path}")
+
+            for name, version in ve.installed_packages:
+                # Skip the editable package since that's the one we're finding dependencies for
+                #
+                # The `virtualenvapi` package also has a bug where comments in the `pip freeze`
```

Comment:
We can remove this check once https://github.com/sjkingo/virtualenv-api/pull/54 gets merged in.